src/opam_query.cppo.ml

```diff
@@ -38,6 +38,32 @@ let guess_archive tag_format opam =
       end
     | _ -> failwith "Unrecognized host specified in `dev-repo:' field."
 
+(* unfortunately, string_of_constraint is local inside OpamFormula *)
+let string_of_constraint (relop, version) =
```

Comment:
Do you think you can ask the opam folks to expose these functions, for anyone else needing them?